### PR TITLE
House Rental App PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this project, @mertkantaroglu, @LeslieAine, and @Zeph180 worked together as 3 people team.

For this project we implemented below requirements;

Our backend repo for the project is [here](https://github.com/LeslieAine/house-rental-back-end.git).

1. The user logs in to the website by using his/her name and email.
2. In the navigation panel, the user can see links to:
   - Houses 
   - Reserve form.
   - My reservations.
   - Add House.
   - Delete House.
3. On the main page, the user can see a list of rental houses.
4. When the user selects a specific house, they can see it with its full description.
   - On the details page, the user can click the "Reserve" button.
5. When the user clicks the "Add House" link in the navigation panel they can see a form for adding a new house.
6. The app became responsive by creating both mobile and desktop versions.
7. When the user clicks the "Delete House" link in the navigation panel they can see a list of all houses with title and "Delete" button.
   - When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.
- To reserve an appointment, the user has to select a date and city.
  - We used the design based on the and add all necessary inputs.
  - The user can also access the "Reserve" page from the navigation panel.
8. When the user clicks the "My Reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date, and city).
9. We added full documentation for our API.

- Fixed linter errors.
- Fixed PR issues.

Looking forward to hearing from you.

Thank you.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
+          <AppRouter />
+        </section>
+      </BrowserRouter>
     </div>
   );
 }
 
 export default App;
+
+function NavBarContainer() {
```
Comment (Meltrust) [inline src/App.js:21]: - [ ]  Please, kindly functions in this file to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.


That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @mertkantaroglu,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  Good readme ✔️ 
-  Beautiful frontend (desktop) ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md [hunk 2]
```diff
@@ -1,70 +1,215 @@
-# Getting Started with Create React App
+<a name="readme-top"></a>
```
[the remark this answers is not in the document]
Comment (mertkantaroglu) [reply README.md:1]: Thank you for suggestions @Meltrust 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @mertkantaroglu,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
#### Hi @mertkantaroglu,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

-  House deletions now work even if the house is reserrved✔️ 
-  Readmes have been corrected✔️ 
-  Improved the mobile nav strategy✔️
-  Debugging code has been removed ✔️ 
-  Functions are now ES6 ✔️ 
-  Removed commented code ✔️ 
-  Splendid job ✔️  



Cheers and Happy coding!👏👏👏


------

**Comment by mertkantaroglu** [comment]
Thanks a lot, @Meltrust for your help and suggestions. We appreciate it! 😄 👍 💯

**Pull request merged**